build_constants.bzl

```diff
@@ -0,0 +1,36 @@
+def _build_constants_impl(repository_ctx):
+    _BUILD_FILE = """
+# DO NOT EDIT: automatically generated for _build_constants rule
+filegroup(
+    name = 'files',
+    srcs = glob(['**']),
+    visibility = ['//visibility:public']
+)
+"""
+    repository_ctx.file("BUILD", _BUILD_FILE, False)
+
+    version = repository_ctx.read(repository_ctx.attr.version_file)
+
+    _CONSTANTS_FILE = """
```

Comment:
is this building version constants for the whole repo? i'm curious to see what this looks like

Reply:
Yep! This is what usage looks like:
https://github.com/player-ui/player/blob/78345ac70c5bb34fcd58147ab49e38f22d3524fd/android/player/BUILD#L2

This is required because the new `maven_publish` rule requires the version to be defined within the `maven_coordinates` tag.

Reply:
Inspiration in thread:
https://github.com/bazelbuild/rules_jvm_external/issues/618

Reply:
now that i'm reading this again, it's not dynamic
does this have the potential to be dynamic for constants that are shared across the platforms or will this always be hard coded

or I guess i'm asking what's the diff between this and creating that output ourselves

Reply:
Hmm, I'm not sure I understand what you mean. It's dynamic in the sense that it makes the version defined in our `VERSION` file accessible at build configuration time. We don't have to change this script every time the version changes. Typically, you're not able to read files during build configuration, only during execution.

The only thing that's hardcoded is the `build_constants` repository rule itself. If we want to add other things, like the `APPLITOOLS_API_KEY`, we could by reading the environment and stuffing into this file. Then you could have access to the key at build configuration time, and maybe even not generate certain test targets if that key doesn't exist.